source/MaterialXCore/Document.cpp

```diff
@@ -605,6 +605,27 @@ void Document::upgradeVersion()
         minorVersion = 36;
     }
 
+    // Upgrade to versions 1.37 and up
+    if ((majorVersion == 1 && minorVersion < 37) || majorVersion < 1)
+    {
+        auto typeToOutputs = [](InterfaceElementPtr interfaceElem)
+        {
+            if (interfaceElem && interfaceElem->hasType())
+            {
+                string type = interfaceElem->getType();
+                if (type != MULTI_OUTPUT_TYPE_STRING)
+                {
+                    interfaceElem->addOutput("out", type);
+                }
+                interfaceElem->removeAttribute(TypedElement::TYPE_ATTRIBUTE);
+            }
+        };
+        for (NodeDefPtr nodeDef : getNodeDefs())
+        {
+            typeToOutputs(nodeDef);
```

Comment:
As far as I can see this lambda is only called from here. If it's only used in one place I think it's better to remove the lambda and just inline the code, as it gives smaller and more readable code.

Reply:
Originally it was also used for nodes before that functionality was deemed no longer needed, I'll remove the lambda.